### PR TITLE
Infix query support

### LINKED ISSUE #99: Infix query language
@smogg cares about people for some reason, and he really wants them to be able to spell `(ip(x) = y)` instead of `(= (:ip x) y)`. So, @lvh has to write an EBNF and strap an extra parser to the compiler.

### PR DESCRIPTION
**Pull request by lvh**
Closes #99.

### CODE REVIEW

**Comment by lvh** [comment]
This builds on top of #102, that should be reviewed and merged first.

File: src/desdemona/query.clj [hunk 1]
```diff
@@ -1,7 +1,11 @@
 (ns desdemona.query
   (:require
    [clojure.core.logic :as l]
-   [clojure.core.match :as m]))
+   [clojure.core.match :as m]
+   [clojure.string :as s]
+   [instaparse.core :as insta]
+   [clojure.java.io :refer [resource]]
+   [taoensso.timbre :refer [spy]]))
```
Comment (sirsean) [inline src/desdemona/query.clj]: Are we using this? I don't see `spy` anywhere else.

**Comment by sirsean** [comment]
I think some of these changes have already been merged. Merge master into this?

**Comment by lvh** [comment]
On it.

File: src/desdemona/query.clj [hunk 2]
```diff
@@ -53,3 +61,24 @@
...
+    (parsed-infix->dsl terms)
+
+    [:eq
+     [:fn-call
+      [:identifier "ip"]
+      [:identifier arg]]
+     [:ipv4-address & addr-parts]]
+    (let [arg (symbol arg)
+          addr (s/join "." addr-parts)]
+      `(~'= (:ip ~arg) ~addr))))
```
Comment (sirsean) [inline src/desdemona/query.clj]: I assume this is just a start, and we'll expand to doing more than IP's at some point later?

Comment (lvh) [inline src/desdemona/query.clj]: Yes. I am not sure which parts of the behavior should live in the grammar and which in this compily bit. I'm inclined to want it to be as early as possible; but writing compiler semantics here is _way_ easier than doing them in the grammar, and the grammar already means that we have a real parser and not a weird machine.

Comment (sirsean) [inline src/desdemona/query.clj]: I think that makes sense. Mostly just commenting on "it seems like we're explicitly only supporting IP addresses here, for now". I don't think we need to change that.

**Comment by lvh** [comment]
The codecov failure is bullshit; it's failing on a partial branch introduced by introducing another match statement.

**Comment by sirsean** [comment]
It seems like coverage doesn't handle those match statements very well.

**Pull request merged**